meta/classes/build_yocto.bbclass

```diff
@@ -0,0 +1,106 @@
+# N.B. this is guaranteed by the layer priority (see layer.conf
+# BBFILE_PRIORITY_xt-core = "5") that this append will
+# run after base_do_unpack and BEFORE any other unpacks
+# such as proprietary code unpack in other layers etc.
+# so, the repo will get updated properly before other
+# related recipes
+build_yocto_repo_sync() {
```

Comment:
bitbake fetcher for repo:// is a part of BB python library, so you cannot override it
and you have to modify BB library which I don't want.
as to Android build, we will probably move some common code for Yocto and Android builds into a common class, but at the moment it seems to be kind of premature optimization